### PR TITLE
Add virtual balance guardrail

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Based on #171 (round up in edge cases). Those rounding changes are all improvements, but an end-to-end test still fails with a panic *before* that rounding code is reached (in `computePriceRange`, because mulDown(Va, Va) -> 0 if Va is too small).

This PR adds a _MIN_VIRTUAL_BALANCE = 1e12 floor and reverts with VirtualBalanceTooLow at the top of computeVirtualBalancesUpdatingPriceRange, to revert with a specific custom error vs numerically.

The guard is intentionally scoped to the out-of-range path (`computeVirtualBalancesUpdatingPriceRange`), because small virtual balances are not in themselves a problem for swaps or liquidity changes (e.g., they can arise from near-complete proportional removal. They only reach the code that can underflow when the pool is out of range, and checking "higher" (e.g., regular `computeCurrentVirtualBalances`, reverts in some valid cases).

We also protect proportional withdrawals, to make it impossible to withdraw down to the point where the pool is unrecoverable (really only possible for single LPs).

To verify that the new end-to-end test catches the actual failure mode, set _MIN_VIRTUAL_BALANCE temporarily to 0. The regression test then fails with panic: division by zero != VirtualBalanceTooLow().

## Type of change

- [X] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [X] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: contracts/lib/ReClammMath.sol [hunk 1]
```diff
@@ -494,6 +508,19 @@ library ReClammMath {
...
+        // "stuck." The check is scoped to this function (and therefore to the out-of-range path) because small virtual
+        // balances are not in themselves a problem for swaps or liquidity changes (e.g., they can arise from near-
+        // complete proportional removal). They only reach the code that can underflow when the pool is out of range.
+        // Note that if this guard triggers, the pool is permanently bricked, since all paths that would update the
+        // virtual balances (including swaps and liquidity changes) would hit this same check and revert. A pool in
+        // this state would need to be placed in recovery mode to allow LPs to exit.
+        if (virtualBalanceA < _MIN_VIRTUAL_BALANCE || virtualBalanceB < _MIN_VIRTUAL_BALANCE) {
+            revert VirtualBalanceTooLow();
+        }
+
```
Comment (joaobrunoah) [inline contracts/lib/ReClammMath.sol:518]: Why here and not after the new virtual balances are computed, in line 410 (end of the function `computeCurrentVirtualBalances`)?

Virtual balances can also go very low when updating price ratio, this check here wouldn't prevent that.

**Review by joaobrunoah — COMMENTED** [review]
It seems to me that the check needs to be in the computeCurrentVirtualBalances function, so that small virtual balances are caught when price ratio is increasing or price range is moving. What do you think?

File: contracts/lib/ReClammMath.sol (hunk 1 above)
Comment (EndymionJkb) [reply contracts/lib/ReClammMath.sol:518]: That's where I put it at first, and why I added the explanation in the comment above. It broke existing tests (specifically, the "remove 99%" liquidity one). The division by zero code only runs when it's out-of-range; putting it in `computeCurrentVirtualBalances` was too broad.

Comment (EndymionJkb) [reply contracts/lib/ReClammMath.sol:518]: testRemoveLiquidity__Fuzz will fail intermittently if you make the guard apply whether it's in or out of range. It's valid to remove ~100% of liquidity proportionally if the pool is in range: but putting the guard "too high" would fail in those legitimate cases. It only blows up when `computePriceRange` is called, when it's out-of-range.

**Comment by EndymionJkb** [comment]
Superseded by #177
